LLama/Native/SafeLlavaModelHandle.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using LLama;
+using LLama.Exceptions;
+
+
+namespace LLama.Native
+{
+    /// <summary>
+    /// A reference to a set of llava model weights
+    /// </summary>
+    public sealed class SafeLlavaModelHandle
+        : SafeLLamaHandleBase
+    {
+
+        internal protected SafeLlavaModelHandle(IntPtr handle)
+            : base(handle, true)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override bool ReleaseHandle()
+        {
+            
+            NativeApi.clip_free(DangerousGetHandle());
+            SetHandle(IntPtr.Zero);
+            return true;
+        }
+
+        /// <summary>
+        /// Load a model from the given file path into memory
+        /// </summary>
+        /// <param name="modelPath"></param>
+        /// <param name="lparams"></param>
+        /// <returns></returns>
+        /// <exception cref="RuntimeError"></exception>
+        public static SafeLlavaModelHandle LoadFromFile(string modelPath, int verbosity )
+        {
+            var ctxContext =  NativeApi.clip_model_load(modelPath, verbosity );            
+            if (ctxContext == IntPtr.Zero)
+                throw new RuntimeError($"Failed to load LLaVa model {modelPath}.");
+
+            return new SafeLlavaModelHandle(ctxContext);
```

Comment:
You can modify `clip_model_load` to directly return `SafeLlavaModelHandle`. That way you never have to directly handle the poniter. See [here](https://github.com/SciSharp/LLamaSharp/blob/master/LLama/Native/SafeLlamaModelHandle.cs#L119) for example.